lib/rex/socket.rb

```diff
@@ -732,7 +732,15 @@ def getlocalname
   # Return peer connection information.
   #
   def getpeername
-    return Socket.from_sockaddr(super)
+    peer_name = nil
+    begin
+      peer_name = Socket.from_sockaddr(super)
+    rescue ::Exception => e
```

Comment:
The exact error I'm trying to rescue is Errno::EINVA, but I can't ```rescue Errno::EINVA``` because I keep getting an initialized error for that. So I elog everything to avoid shutting up errors.